package/src/commandHighlighter.ts

```diff
@@ -31,7 +32,7 @@ export default class KustoCommandHighlighter implements monaco.editor.IEditorCon
     private highlightCommandUnderCursor(changeEvent: monaco.editor.ICursorSelectionChangedEvent): void {
             // Looks like the user selected a bunch of text. we don't want to highlight the entire command in this case - since highlighting
             // the text is more helpful.
-            if (!changeEvent.selection.isEmpty()) {
+            if (!changeEvent.selection.isEmpty() || this.editor.getModel().getModeId() !== 'kusto') {
```

Comment:
i would keep `highlightCommandunderCursor` clean and move this check to the `onDidChangecursosSelection` . it may catch more future cases that happen outside this specific funtion